### PR TITLE
feat(SPV-960): Replaced any with types

### PR DESCRIPTION
**Pull request by Nazarii-4chain**
# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: src/components/ContactsTabContent/ContactsTabContent.tsx [hunk 1]
```diff
@@ -12,21 +12,20 @@ import {
...
   DataTable,
   NoRecordsText,
-  TransactionEditDialogProps,
 } from '@/components';
 import { ContactExtended } from '@/interfaces/contacts.ts';
 
 export interface ContactsTabContentProps {
   contacts: ContactExtended[];
-  EditDialog?: React.ComponentType<ContactEditDialogProps | TransactionEditDialogProps>;
+  ContactEditDialog?: React.ComponentType<ContactEditDialogProps>;
```
Comment (swierzbicki) [inline src/components/ContactsTabContent/ContactsTabContent.tsx:20]: Is this change is related with changing `any` types to specific types? 🤔

File: src/components/ViewDialog/ViewDialog.tsx [hunk 2]
```diff
@@ -46,7 +43,7 @@ export const ViewDialog = ({ row }: ViewDialogProps) => {
       <DialogTrigger className="w-full">
         <DropdownMenuItem onSelect={(e) => e.preventDefault()}>View</DropdownMenuItem>
       </DialogTrigger>
-      <DialogContent className="max-w-3xl max-h-5/6 overflow-auto">
+      <DialogContent className="max-w-3xl max-h-[85vh] overflow-auto">
```
Comment (swierzbicki) [inline src/components/ViewDialog/ViewDialog.tsx:46]: Is this change is related with changing `any` types to specific types? 🤔

File: src/logger/index.ts [hunk 3]
```diff
@@ -13,15 +13,10 @@ const levels: { [key: number]: string } = {
 // Type for log objects
 type LogObject = {
   level?: number | string;
-  // TODO [explicit-any]: consider add type if it's possible
-  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
-  [key: string]: any;
+  [key: string]: unknown;
```
Comment (swierzbicki) [inline src/logger/index.ts:16]: Do we really don't know what value we could have here?

**Review by swierzbicki — CHANGES_REQUESTED** [review]
1.Added some comments
2. Code doesn't compile

File: src/components/ContactAcceptDialog/ContactAcceptDialog.tsx [hunk 4]
```diff
@@ -17,14 +17,13 @@ import {
...
 
 export interface ContactAcceptDialogProps {
-  // TODO [explicit-any]: consider add type if it's possible
-  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
-  row: Row<any>;
+  row: Row<Contact>;
 }
 
-export const ContactAcceptDialog = ({ row }: ContactAcceptDialogProps) => {
+export function ContactAcceptDialog({ row }: ContactAcceptDialogProps) {
```
[the remark this answers is not in the document]
Comment (Nazarii-4chain) [reply src/components/ContactAcceptDialog/ContactAcceptDialog.tsx]: I was experimenting. Reverted this change

File: src/components/ContactsTabContent/ContactsTabContent.tsx (hunk 1 above)
Comment (Nazarii-4chain) [reply src/components/ContactsTabContent/ContactsTabContent.tsx:20]: Yes

File: src/components/ViewDialog/ViewDialog.tsx (hunk 2 above)
Comment (Nazarii-4chain) [reply src/components/ViewDialog/ViewDialog.tsx:46]: Found this issue while checking types here. This change does not relate to `any` issue

File: src/logger/index.ts (hunk 3 above)
Comment (Nazarii-4chain) [reply src/logger/index.ts:16]: Almost everything can be here. From undefined to custom types.

File: src/components/ContactsTabContent/ContactsTabContent.tsx (hunk 1 above)
Comment (dorzepowski) [reply src/components/ContactsTabContent/ContactsTabContent.tsx:20]: how?

File: src/components/DataTable/DataTable.tsx [hunk 5]
```diff
@@ -32,28 +33,38 @@ import {
...
+
+export type RowType = XPub | Contact | AccessKey | Destination | PaymailAddress | Tx;
 
 interface DataTableProps<TData, TValue> {
   columns: ColumnDef<TData, TValue>[];
   data: TData[];
-  EditDialog?: React.ComponentType<ContactEditDialogProps | TransactionEditDialogProps | DestinationEditDialogProps>;
+  ContactEditDialog?: React.ComponentType<ContactEditDialogProps>;
+  TransactionEditDialog?: React.ComponentType<TransactionEditDialogProps>;
+  DestinationEditDialog?: React.ComponentType<DestinationEditDialogProps>;
```
Comment (dorzepowski) [inline src/components/DataTable/DataTable.tsx:47]: Why this change? it looks ugly and cumbersome.

File: src/components/DataTable/DataTable.tsx [hunk 6]
```diff
@@ -112,10 +123,13 @@ export function DataTable<TData, TValue>({
...
-                      {EditDialog ? <EditDialog row={row} /> : null}
-                      {DeleteDialog ? <DeleteDialog row={row} /> : null}
-                      {RevokeKeyDialog ? <RevokeKeyDialog row={row} /> : null}
+                      <ViewDialog row={row as Row<RowType>} />
+                      {ContactEditDialog ? <ContactEditDialog row={row as Row<Contact>} /> : null}
+                      {TransactionEditDialog ? <TransactionEditDialog row={row as Row<Tx>} /> : null}
+                      {DestinationEditDialog ? <DestinationEditDialog row={row as Row<Destination>} /> : null}
+                      {DeleteDialog ? <DeleteDialog row={row as Row<Contact>} /> : null}
+                      {PaymailDeleteDialog ? <PaymailDeleteDialog row={row as Row<PaymailAddress>} /> : null}
+                      {RevokeKeyDialog ? <RevokeKeyDialog row={row as Row<AccessKey>} /> : null}
```
Comment (dorzepowski) [inline src/components/DataTable/DataTable.tsx:132]: ohhh it looks hmm.. not the best.

IMHO it would be much much better if this component would accept 2 optional components:
1. `inlineActions` which would define such buttons as Accept and Decline in contacts
2. `actions` which would define actions menu items and the logic how it should handle it.

With the proper way of using generics it should be easy to pass the row in a valid way to any action component.

File: src/components/ViewDialog/ViewDialog.tsx (hunk 2 above)
Comment (dorzepowski) [reply src/components/ViewDialog/ViewDialog.tsx:46]: what is this change? I don't understand this change, I would suggest to extract it into separated commit an PR

File: src/logger/index.ts [hunk 7]
```diff
@@ -48,6 +43,7 @@ const pinoLogger: Logger = pino({
         console.warn(getDefaultObject(o));
       },
       error: (o: LogObject = {}) => {
+        console.log('o');
```
Comment (dorzepowski) [inline src/logger/index.ts:46]: ??

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: src/components/ContactsTabContent/ContactsTabContent.tsx (hunk 1 above)
Comment (Nazarii-4chain) [reply src/components/ContactsTabContent/ContactsTabContent.tsx:20]: The `any` issue was deeper in the code. This change is needed to handle `any` issue

File: src/components/ViewDialog/ViewDialog.tsx (hunk 2 above)
Comment (Nazarii-4chain) [reply src/components/ViewDialog/ViewDialog.tsx:46]: It changed the max height of DialogContent in ViewDialog component to 85vh. Will remove this

File: src/components/DataTable/DataTable.tsx (hunk 5 above)
Comment (Nazarii-4chain) [reply src/components/DataTable/DataTable.tsx:47]: Related to `any` issue. We can have one complex component with bunch of `if else` statements that will be hard to read and make changes even harder. Or have a few smaller components with straightforward logic.

File: src/components/DataTable/DataTable.tsx [hunk 8]
```diff
@@ -28,33 +17,33 @@ import {
...
 
 interface DataTableProps<TData, TValue> {
   columns: ColumnDef<TData, TValue>[];
   data: TData[];
-  EditDialog?: React.ComponentType<ContactEditDialogProps | TransactionEditDialogProps | DestinationEditDialogProps>;
-  AcceptDialog?: React.ComponentType<ContactAcceptDialogProps>;
-  DeleteDialog?: React.ComponentType<ContactDeleteDialogProps>;
-  RejectDialog?: React.ComponentType<ContactRejectDialogProps>;
-  RevokeKeyDialog?: React.ComponentType<RevokeKeyDialogProps>;
+  renderItem?: (row: Row<RowType>) => React.ReactNode;
```
Comment (dorzepowski) [inline src/components/DataTable/DataTable.tsx:41]: Use generics from the interface not some custom type, this way you won't need to use type assertions later in code.
```suggestion
  renderItem?: (row: Row<TData>) => React.ReactNode;
```

File: src/components/DataTable/DataTable.tsx [hunk 9]
```diff
@@ -112,10 +105,10 @@ export function DataTable<TData, TValue>({
                     </DropdownMenuTrigger>
                     <DropdownMenuContent align="end">
                       <DropdownMenuLabel>Actions</DropdownMenuLabel>
-                      <ViewDialog row={row} />
-                      {EditDialog ? <EditDialog row={row} /> : null}
-                      {DeleteDialog ? <DeleteDialog row={row} /> : null}
-                      {RevokeKeyDialog ? <RevokeKeyDialog row={row} /> : null}
+                      <ViewDialog row={row as Row<RowType>} />
+                      {renderItem ? renderItem(row as Row<RowType>) : null}
```
Comment (dorzepowski) [inline src/components/DataTable/DataTable.tsx:109]: This is better 
but still imho there should be 2 functions:
`renderActions` which can handle that the row is contact and what actions can be applied to it
and `renderInlineActions`

File: src/components/DataTable/DataTable.tsx [hunk 10]
```diff
@@ -98,8 +87,12 @@ export function DataTable<TData, TValue>({
                 <TableCell>
                   {table.getColumn('status') && row.getValue('status') === ContactStatus.Awaiting ? (
                     <div className="grid grid-cols-2 items-center w-fit gap-4 ">
-                      {AcceptDialog ? <AcceptDialog row={row} /> : null}
-                      {RejectDialog ? <RejectDialog row={row} /> : null}
+                      {isContact(row.original) && row.original.status === ContactStatus.Awaiting && (
+                        <>
+                          <ContactAcceptDialog row={row as Row<Contact>} />
+                          <ContactRejectDialog row={row as Row<Contact>} />
```
Comment (dorzepowski) [inline src/components/DataTable/DataTable.tsx:93]: This should be replaced with something similar to renderItem, for example `renderInlineActions`

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: src/components/DataTable/DataTable.tsx (hunk 10 above)
Comment (Nazarii-4chain) [reply src/components/DataTable/DataTable.tsx]: Added a prop

File: src/components/DataTable/DataTable.tsx [hunk 11]
```diff
@@ -85,16 +85,17 @@ export function DataTable<TData, TValue>({ columns, data, renderItem }: DataTabl
...
+                  {/*{table.getColumn('status') && row.getValue('status') === ContactStatus.Awaiting ? (*/}
+                  {/*  <div className="grid grid-cols-2 items-center w-fit gap-4 ">*/}
+                  {/*    {isContact(row.original) && row.original.status === ContactStatus.Awaiting && (*/}
+                  {/*      <>*/}
+                  {/*        <ContactAcceptDialog row={row as Row<Contact>} />*/}
+                  {/*        <ContactRejectDialog row={row as Row<Contact>} />*/}
+                  {/*      </>*/}
+                  {/*    )}*/}
+                  {/*  </div>*/}
+                  {/*) : null}*/}
```
Comment (dorzepowski) [inline src/components/DataTable/DataTable.tsx:97]: Please remove commented out code

**Review by dorzepowski — APPROVED** [review]

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: src/components/DataTable/DataTable.tsx [hunk 12]
```diff
@@ -96,12 +85,17 @@ export function DataTable<TData, TValue>({
...
+                  {renderInlineItem ? renderInlineItem(row) : null}
```
[the remark this answers is not in the document]
Comment (Nazarii-4chain) [reply src/components/DataTable/DataTable.tsx]: Removed

**Review by dorzepowski — APPROVED** [review]

**Review by swierzbicki — APPROVED** [review]

**Review by chris-4chain — APPROVED** [review]
